lib/logstash/filters/aggregate.rb

```diff
@@ -317,13 +348,31 @@ def filter(event)
       # retrieve the current aggregate map
       aggregate_maps_element = @@aggregate_maps[task_id]
       
+
       # create aggregate map, if it doesn't exist
       if (aggregate_maps_element.nil?)
         return if @map_action == "update"
         # create new event from previous map, if @push_previous_map_as_event is enabled
         if (@push_previous_map_as_event and !@@aggregate_maps.empty?)
           previous_map = @@aggregate_maps.shift[1].map
           event_to_yield = LogStash::Event.new(previous_map)
+
+          if @timeout_task_id_field
+            event_to_yield[@timeout_task_id_field] = task_id
+          end
+
+          # Call code block if available
+          if @timeout_code
+            begin
+              @timeout_codeblock.call(event)
+              events_to_flush << event
```

Comment:
Up to me, in filter method, there are no events to flush.
So this line has to be removed.
Do you agree with that ?